optuna/trial.py

```diff
@@ -320,13 +373,29 @@ def suggest_categorical(self, name, choices):
             Suggest a kernel function of `SVC <https://scikit-learn.org/stable/modules/generated/
             sklearn.svm.SVC.html>`_.
 
-            .. code::
+            .. testcode::
+
+                import optuna
+                import numpy as np
+                from sklearn.svm import SVC
+                from sklearn.model_selection import train_test_split
+
+                def objective(trial):
+                    kernel = trial.suggest_categorical('kernel', ['linear', 'poly', 'rbf'])
+                    assert kernel in ['linear', 'poly', 'rbf']
+
+                    np.random.seed(seed=0)
+                    X = np.random.randn(50).reshape(-1, 1)
+                    y = np.random.randint(0, 2, 50)
+                    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
```

Comment:
Please hide this dataset preparation, as I suggested in the `trial.suggest_loguniform` example.